web-frontend/src/main.tsx

```diff
@@ -50,8 +51,9 @@ const router = createBrowserRouter(
       <Route path=":nickname/send-coins" element={<SendCoins />} />
       <Route path=":nickname/receive" element={<ReceiveCoins />} />
       <Route path=":nickname/assets" element={<></>} />
-      <Route path=":nickname/settings" element={<Settings />} />
       <Route path=":nickname/contacts" element={<></>} />
+      <Route path=":nickname/settings" element={<Settings />} />
+      <Route path=":nickname/settings/update" element={<SettingsEdit />} />
```

Comment:
why do we need a new page?

Reply:
We need to change the route to update the nickname in the URL (at least for now).
The page we are adding has the edit button that redirects when is success.

BTW, It's pretty fluid and match with the first design.
We can check that together if you prefer!